fix freebsd compilation error

I have compilation error on FreeBSD 14.1 when run `pip install mitmproxy`.  mitmproxy depends on mitmproxy-rs which depends on boringtun.

OS: FreeBSD 14.1-RELEASE-p5
Arch: amd64
Rustc: rustc 1.81.0 (f54dd915b 2024-09-02) 
         

> Compiling boringtun v0.6.0 (https://github.com/cloudflare/boringtun?rev=e3252d9c4f4c8fc628995330f45369effd4660a1#e3252d9c)
>       error[E0599]: no associated item named `CLOCK_BOOTTIME` found for struct `ClockId` in the current scope
>        --> /home/user/.cargo/git/checkouts/boringtun-379ccad4d4acd372/e3252d9/boringtun/src/sleepyinstant/unix.rs:9:36
>       9 | const CLOCK_ID: ClockId = ClockId::CLOCK_BOOTTIME;
>         |                                    ^^^^^^^^^^^^^^ associated item not found in `ClockId`
>         |
>       help: there is an associated constant `CLOCK_UPTIME` with a similar name
>         |
>       9 | const CLOCK_ID: ClockId = ClockId::CLOCK_UPTIME;

Related issue: mitmproxy/mitmproxy#6378,  #386 #29.

--- a/boringtun/src/sleepyinstant/unix.rs
+++ b/boringtun/src/sleepyinstant/unix.rs
@@ -3,9 +3,9 @@ use std::time::Duration;
 use nix::sys::time::TimeSpec;
 use nix::time::{clock_gettime, ClockId};
 
-#[cfg(any(target_os = "macos", target_os = "ios", target_os = "tvos"))]
+#[cfg(any(target_os = "macos", target_os = "ios", target_os = "tvos", target_os = "freebsd"))]
 const CLOCK_ID: ClockId = ClockId::CLOCK_MONOTONIC;
-#[cfg(not(any(target_os = "macos", target_os = "ios", target_os = "tvos")))]
+#[cfg(not(any(target_os = "macos", target_os = "ios", target_os = "tvos", target_os = "freebsd")))]
 const CLOCK_ID: ClockId = ClockId::CLOCK_BOOTTIME;
 
 #[derive(Clone, Copy, Debug)]

--- a/boringtun/src/sleepyinstant/unix.rs
+++ b/boringtun/src/sleepyinstant/unix.rs
@@ -3,9 +3,19 @@ use std::time::Duration;
 use nix::sys::time::TimeSpec;
 use nix::time::{clock_gettime, ClockId};
 
-#[cfg(any(target_os = "macos", target_os = "ios", target_os = "tvos", target_os = "freebsd"))]
+#[cfg(any(
+    target_os = "macos",
+    target_os = "ios",
+    target_os = "tvos",
+    target_os = "freebsd"
+))]
 const CLOCK_ID: ClockId = ClockId::CLOCK_MONOTONIC;
-#[cfg(not(any(target_os = "macos", target_os = "ios", target_os = "tvos", target_os = "freebsd")))]
+#[cfg(not(any(
+    target_os = "macos",
+    target_os = "ios",
+    target_os = "tvos",
+    target_os = "freebsd"
+)))]
 const CLOCK_ID: ClockId = ClockId::CLOCK_BOOTTIME;
 
 #[derive(Clone, Copy, Debug)]

